### PR TITLE
Refactor IQueryResult usage

### PR DESCRIPTION
**Pull request by Skrypt**
Moving LuceneQueryResult to OrchardCore.Lucene.Abstractions
Moving SqlQueryResult to OrchardCore.Data.Abstractions
Adding OrchardCore.Queries.Abstractions to both these projects to use IQueryResults interface.

### CODE REVIEW

**Review by jtkech — APPROVED** [review]
LGTM

**Comment by hishamco** [comment]
Looks great, but this will be considered as binary breaking changes

**Comment by Skrypt** [comment]
We would need to keep LuceneQueryResults and mark it as obsolete.

File: src/OrchardCore/OrchardCore.Queries.Abstractions/LuceneQueryResults.cs [hunk 1]
```diff
@@ -1,7 +1,10 @@
+using System;
 using System.Collections.Generic;
+using OrchardCore.Queries;
 
 namespace OrchardCore.Queries
 {
+    [Obsolete("Moved to OrchardCore.Lucene.Abstractions", false)]
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Queries.Abstractions/LuceneQueryResults.cs:7]: This class has been deprecated and we will be removed in the major release, please use OrchardCore.Lucene.Abstractions instead.

**Pull request merged**